includes/components/class-attribute.php

```diff
@@ -1179,7 +1179,11 @@ public function add_sort_options( $form_args, $form ) {
 		$form_args['fields']['_sort']['options'] = array_merge( $form_args['fields']['_sort']['options'], $options );
 
 		// Set default option.
-		$default = get_option( hp\prefix( $model . '_default_order' ) );
+		if ( is_tax( hp\prefix( $model . '_category' ) ) ) {
```

Comment:
This one seems to fire on the category pages, but not if you search by category (via the filters form), it should also work if you view any of the child categories. You can try using the get_category_id helper, it gets the current category ID depending if it's a category or search page, and if the ID is not empty you can use get_ancestors function (needs testing, maybe there's a more simple way, caching this also may be a good idea) to get the parent terms until it has the sort order set, and if even the top one doesn't have it then the default one should be used (so you can move it from "else" back, then it'll be the default value)

Reply:
As discussed there is the problem is that when the user changes a category in the filter form, then in GET request uses data from the sorting form, and then the value of the sorting form is get from the GET request, not from our filter hook to the sorting form

Reply:
In which way it need to be fixed?

Reply:
This should work for both 2 cases:

- Searching by a category via the filter form
- Viewing the category page

In both cases the get_category_id helper in this component should return the category ID so you can try using it. Also the default sorting option should be used for any child categories of the category where the default option is selected (so it should inherit these settings), please check the first comment it seems to be valid (using get_category_id and then get_ancestors to find the "nearest" parent with the default sorting option setting if defined).